From 6e435911394b05c91b92d4c332c455ec569e22fb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20Wei=C3=9Fschuh?= <thomas.weissschuh@linutronix.de>
Date: Tue, 7 Jul 2026 08:07:14 +0200
Subject: [PATCH 03015/15333] timekeeping: Fold vdso_time_update_aux()
 declarations into the generic ifdeffery
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The only caller of vdso_time_update_aux() is already gated behind
CONFIG_POSIX_AUX. The additional check in the header files is not
necessary.

Remove it and then fold the declarations into the existing
CONFIG_GENERIC_GETTIMEOFDAY ifdeffery.

Signed-off-by: Thomas Weißschuh <thomas.weissschuh@linutronix.de>
Signed-off-by: Thomas Gleixner <tglx@kernel.org>
Link: https://patch.msgid.link/20260707-timekeeping-header-cleanup-v1-1-e85ad96409a9@linutronix.de
---
 include/linux/timekeeper_internal.h | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/include/linux/timekeeper_internal.h b/include/linux/timekeeper_internal.h
index 4486dfd5d0de..58c83d6e65b3 100644
--- a/include/linux/timekeeper_internal.h
+++ b/include/linux/timekeeper_internal.h
@@ -194,6 +194,7 @@ struct timekeeper {
 
 extern void update_vsyscall(struct timekeeper *tk);
 extern void update_vsyscall_tz(void);
+extern void vdso_time_update_aux(struct timekeeper *tk);
 
 #else
 
@@ -203,12 +204,9 @@ static inline void update_vsyscall(struct timekeeper *tk)
 static inline void update_vsyscall_tz(void)
 {
 }
-#endif
-
-#if defined(CONFIG_GENERIC_GETTIMEOFDAY) && defined(CONFIG_POSIX_AUX_CLOCKS)
-extern void vdso_time_update_aux(struct timekeeper *tk);
-#else
-static inline void vdso_time_update_aux(struct timekeeper *tk) { }
+static inline void vdso_time_update_aux(struct timekeeper *tk)
+{
+}
 #endif
 
 #endif /* _LINUX_TIMEKEEPER_INTERNAL_H */
-- 
2.55.0

